WordPress/Classes/Stores/StatsWidgetsStore.swift

```diff
@@ -0,0 +1,224 @@
+            }
+        }
+
+        if let newAllTimeData = refreshStats(type: HomeWidgetAllTimeData.self) {
+            HomeWidgetAllTimeData.write(items: newAllTimeData)
+
+            if #available(iOS 14.0, *) {
+                WidgetCenter.shared.reloadTimelines(ofKind: WPHomeWidgetAllTimeKind)
+            }
+        }
+    }
+
+    /// Initialize the local cache for widgets, if it does not exist
+    func initializeStatsWidgetsIfNeeded() {
+        guard #available(iOS 14.0, *) else {
+            return
+        }
+        if HomeWidgetTodayData.read() == nil {
+            DDLogInfo("StatsWidgets: Writing initialization data into HomeWidgetTodayData.plist")
+            HomeWidgetTodayData.write(items: initializeHomeWidgetData(type: HomeWidgetTodayData.self))
+            WidgetCenter.shared.reloadTimelines(ofKind: WPHomeWidgetTodayKind)
+        }
+
+        if HomeWidgetAllTimeData.read() == nil {
+            DDLogInfo("StatsWidgets: Writing initialization data into HomeWidgetAllTimeData.plist")
+            HomeWidgetAllTimeData.write(items: initializeHomeWidgetData(type: HomeWidgetAllTimeData.self))
+            WidgetCenter.shared.reloadTimelines(ofKind: WPHomeWidgetAllTimeKind)
+        }
+    }
+
+    /// Store stats in the widget cache
+    /// - Parameters:
+    ///   - widgetType: concrete type of the widget
+    ///   - stats: stats to be stored
+    func storeHomeWidgetData<T: HomeWidgetData>(widgetType: T.Type, stats: Codable) {
+        guard #available(iOS 14.0, *),
+              let siteID = SiteStatsInformation.sharedInstance.siteID else {
+            return
+        }
+
+        var homeWidgetCache = T.read() ?? initializeHomeWidgetData(type: widgetType)
+        guard let oldData = homeWidgetCache[siteID.intValue] else {
+            DDLogError("StatsWidgets: Failed to find a matching site")
+            return
+        }
+        let blogService = BlogService(managedObjectContext: ContextManager.shared.mainContext)
+
+        guard let blog = blogService.blog(byBlogId: siteID) else {
+            DDLogError("StatsWidgets: the site does not exist anymore")
+            // if for any reason that site does not exist anymore, remove it from the cache.
+            homeWidgetCache.removeValue(forKey: siteID.intValue)
+            T.write(items: homeWidgetCache)
+            return
+        }
+        var widgetKind = ""
+        if widgetType == HomeWidgetTodayData.self, let stats = stats as? TodayWidgetStats {
+
+            widgetKind = WPHomeWidgetTodayKind
+
+            homeWidgetCache[siteID.intValue] = HomeWidgetTodayData(siteID: siteID.intValue,
+                                                                   siteName: blog.title ?? oldData.siteName,
+                                                                   url: blog.url ?? oldData.url,
+                                                                   timeZone: blogService.timeZone(for: blog),
```

Comment:
all the calls to `blogService.timeZone(for: blog)` should be changed to `blog.timeZone`